### PR TITLE
Feature/photo likes

### PR DESCRIPTION
**Pull request by LukaNikcevic77**
Pr zbog backend-a koji ne ucitava odma podatke vjerovatno.

### CODE REVIEW

File: src/components/UserCard/index.js
Comment (tonkec) [inline src/components/UserCard/index.js]: please be careful about committing development code

File: src/context/realTimeDataTrackerContext.js
Comment (tonkec) [inline src/context/realTimeDataTrackerContext.js]: why is this context called like that?

File: src/pages/ProfilePage/PhotoGallery/Downvote.js
Comment (tonkec) [inline src/pages/ProfilePage/PhotoGallery/Downvote.js]: remove console.log

File: src/pages/ProfilePage/PhotoGallery/Upvote.js
Comment (tonkec) [inline src/pages/ProfilePage/PhotoGallery/Upvote.js]: console.log

**Review by tonkec — CHANGES_REQUESTED** [review]

**Comment by tonkec** [comment]
Couple of things are missing
- user should see an error when they try to like/dislike a photo that is already liked/disliked
- user can not like their own photos, so hide like/unlike button from profile page
- dislike and like functions in `PhotoService` should return the `data` object, then you will not get the `undefined` any more
- user can not dislike photo that has not been already liked. So photo can not have -1 likes

**Comment by tonkec** [comment]
Can you please elaborate why did you choose context instead of redux?

**Comment by tonkec** [comment]
Can you also please declutter code from unused vars?

File: src/context/realTimeDataTrackerContext.js
Comment (LukaNikcevic77) [reply src/context/realTimeDataTrackerContext.js]: because it will share like count and other stats important to have to all components that will be updated from different places on change which makes it basically real time traceker.

**Comment by LukaNikcevic77** [comment]
> Can you please elaborate why did you choose context instead of redux?

Because I have been using context since the first day of react and never really used redux, so it never crossed my mind to go for redux libary, if we are allready using react context.

File: src/components/UserCard/index.js
Comment (tonkec) [inline src/components/UserCard/index.js]: @LukaNikcevic77 can you please remove these dummy console logs? Be careful about committing this type of code

File: src/pages/ProfilePage/PhotoGallery/Downvote.js
Comment (tonkec) [inline src/pages/ProfilePage/PhotoGallery/Downvote.js:5]: I can dislike an image after I liked it. Can you please check it out? The dislike button is disabled when I like an image

Screen Shot 2023-09-15 at 10 05 37

**Review by tonkec — CHANGES_REQUESTED** [review]
- there are some bugs that need to be resolved before merging. I explained in the comments
- be careful how you name files, we can't have files name with word `logic` in it. I renamed it now, but for future reference I will leave it to you to think of a better name
- there were plenty of unused vars, can you be more careful about that, it is making our code ugly. I removed them all now.
- can you please commit more often, smaller steps with more meaningful commit messages? Can you read this https://www.aleksandrhovhannisyan.com/blog/atomic-git-commits/
- I would love to see at least one passing unit test for likes/dislikes, do you think you can do that or should I do it?
- can you also resolve conflicts with the master branch

**Review by tonkec — APPROVED** [review]

**Pull request merged**
